profiles/scraper.py

```diff
@@ -388,13 +415,33 @@ def parse_profile_experiences(self, company, role):
         :return: List of dictionaries containing information about each experience. Return None if no experience.
         """
         experiences = []
-        exps = self.get_multiple_by_xpath_or_none('EXPERIENCES')
+        exps = self.get_multiple_by_xpath_or_none('EXPERIENCES', wait=True)
         if not exps:
             return None
 
         experience_with_incorrect_company = 0
         for exp in exps:
             temp = {}
+            # checking if this experience is "multiple experience under 1 company"
+            # since for multiple experiences, the fields are slightly different than single experience
+            # we distinguish these experiences by checking if "Company Name" exists in the company name field
+    
+            company_name = self.element_get_text_by_xpath_or_none(exp, 'COMPANY_MULTI_EXPERIENCES')
+            if "Company Name" in company_name and not self.get_multiple_by_xpath_or_none('MULTI_EXPERIENCES'): # this means this section has multiple experiences in 1 company
+                multi_exps = self.get_multiple_by_xpath_or_none('MULTI_EXPERIENCES')
+                # need to iterate thru the li elements of these experiences and add into temp (make sure temp['company'] is there for all experiences
+                for experience in multi_exps:
+                    temp['company'] = company_name.split('\n')[1]
+                    title = self.element_get_text_by_xpath_or_none(
+                        experience, 'TITLE_MULTI_EXPERIENCES') if self.element_get_text_by_xpath_or_none(experience, 'TITLE_MULTI_EXPERIENCES') is not None else ""
+                    temp['title'] = title.split('\n')[1]
+                    temp['dates'] = self.element_get_text_by_xpath_or_none(
+                        experience, 'DATE_RANGE').split('\n')[1] if self.element_get_text_by_xpath_or_none(experience, 'DATE_RANGE') is not None else ""
+
+                    if temp['company'].lower() != company.lower():
+                        experience_with_incorrect_company +=1
+                    experiences.append(temp)
```

Comment:
Is this similar to the below extraction? if so move it into a function call

Reply:
it's not, because for this part is for multiple experiences under 1 company, and for the fields in these experiences, the actual title/company name/dates are in the second line of the text that we extract. That's why you see things like title.split('\n')[1]. That's not the case for single experience at a single company.